src/local/position/extra_math.rs

```diff
@@ -55,9 +55,49 @@ impl Position {
     /// );
     /// ```
     #[inline]
+    #[track_caller]
     pub fn offset(&mut self, x: i32, y: i32) {
         *self = *self + (x, y);
     }
+
+    /// Adds an `(x, y)` pair to this room position's world coordinates and
+    /// returns the result.
+    ///
+    /// Will change rooms if necessary.
+    ///
+    /// # Errors
+    /// Returns `Err` if the new position's room is outside bounds.
+    ///
+    /// For a panicking variant of this function, see [`Position::add`].
+    ///
+    /// See [`Position::from_world_coords`].
+    #[inline]
+    pub fn checked_add(self, rhs: (i32, i32)) -> Result<Position, WorldPositionOutOfBoundsError> {
+        let (x1, y1) = self.world_coords();
+        let (x2, y2) = rhs;
+
+        Position::checked_from_world_coords(x1 + x2, y1 + y2)
```

Comment:
`x1.checked_add(x2).ok_or(...)?`, here and elsewhere we directly take `(i32, i32)` pairs.